src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/ProxyStreamFactory.java

```diff
@@ -1434,7 +1435,8 @@ private boolean cache(BeginFW begin)
             final HttpBeginExFW httpBeginEx = extension.get(httpBeginExRO::wrap);
             final ListFW<HttpHeaderFW> responseHeaders = httpBeginEx.headers();
             final boolean isCacheable = HttpCacheUtils.isPublicCacheableResponse(responseHeaders);
-            if (isCacheable)
+            final ListFW<HttpHeaderFW> requestHeaders = streamCorrelation.requestHeaders(pendingRequestHeadersRO);
+            if (isCacheable && canStore(requestHeaders))
```

Comment:
Should the `canStore` method receive `requestHeaders` or `responseHeaders` as a parameter?

This call site passes `requestHeaders`, but the method body receives `responseHeaders`.

Reply:
For the case @stanculescu, it should be request headers because it is a request cache directive she is implementing.  However, in the future we will need to do both request and response headers.  Maybe we can implement the check for both now, as it seems like it would be the exact same spot in the code and same check, just on different headers :-)